CAIPs/caip-372.md

```diff
@@ -0,0 +1,130 @@
+---
+caip: 372
+title: Wallet Information Metadata Standard
+author: Pedro Gomes (@pedrouid)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/372
+status: Draft
+type: Standard
+created: 2025-08-11
+---
+
+## Simple Summary
+
+This CAIP defines a standard `WalletInfo` metadata object to identify and validate the wallet provider that a client application is interacting with.
+It can be used across RPC methods, APIs, and protocols to provide consistent, cross-chain wallet identification.
+
+## Abstract
+
+The `WalletInfo` object provides a minimal, standardized set of fields that uniquely identify a wallet provider and allow dApps to confirm they are connected to the intended wallet.
+The object is designed for use in [CAIP-25] session properties, [CAIP-282] messages, and other API contexts.
```

Comment:
> for use in [CAIP-25] session properties

tbh i got a little confused by whether you meant properties of a caip-25 session or sessionProperties specifically... and then went down the rabbithole of what situations would justify dapps sending the WalletInfo object to a wallet (tell me if you're not this wallet? correct this if you've upgraded since last session?)